modules/core/src/test/scala/org/tessellation/infrastructure/trust/TrustModelSuite.scala

```diff
@@ -0,0 +1,33 @@
+package org.tessellation.infrastructure.trust
+
+import cats.syntax.applicative._
+
+import org.tessellation.infrastructure.trust.TrustModel.calculateTrust
+import org.tessellation.infrastructure.trust.generators.{genPeerPublicTrust, genPeerTrustInfo}
+import org.tessellation.schema.generators.peerIdGen
+import org.tessellation.sdk.domain.trust.storage.{PublicTrustMap, TrustMap}
+
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object TrustModelSuite extends SimpleIOSuite with Checkers {
+  test("peer labels contain undiscovered peers does not throw an exception") {
+    val gen = for {
+      trust <- Gen.mapOfN(4, genPeerTrustInfo)
+      labels <- Gen.mapOfN(4, genPeerPublicTrust)
+      publicTrustMap = PublicTrustMap(labels)
+      peerId <- peerIdGen
+    } yield (TrustMap(trust, publicTrustMap), peerId)
+
+    forall(gen) {
+      case (trust, selfId) =>
+        for {
+          trustKeySet <- trust.trust.keySet.pure
+          labelKeySet = trust.peerLabels.value.keySet
+          _ = calculateTrust(trust, selfId) // expect to not throw an exception
+        } yield expect.all(labelKeySet.diff(trustKeySet).nonEmpty)
```

Comment:
the test specifies `does not throw an exception` but there is no assertion checking that. instead, you check if the diff between labels and trusts is non empty. either change the test name to match what you're testing or change the assertion to check what's written in the test.

If you want to check the method throws or not, you can use `.attempt` on `IO` which maps `IO[A]` to `IO[Either[Throwable, A]]`, so you can write an explicit assertion (is left or right side of either). (it allows also to check for a specific error to be thrown)

Reply:
Added the explicit `expect`.